src/charm.py

```diff
@@ -181,6 +181,94 @@ def _on_synapse_nginx_pebble_ready(self, _: ops.HookEvent) -> None:
+
+        Returns:
+            admin access token or None if fails.
+        """
+        peer_relation = self._get_peer_relation()
+        if not peer_relation:
+            logger.error(
+                "Failed to get admin access token: no peer relation %s found", PEER_RELATION_NAME
+            )
+            return None
+        admin_access_token = None
+        if JUJU_HAS_SECRETS:
+            secret_id = peer_relation.data[self.app].get(SECRET_ID)
+            if secret_id:
+                try:
+                    secret = self.model.get_secret(id=secret_id)
+                    admin_access_token = secret.get_content().get(SECRET_KEY)
+                    return admin_access_token
+                except ops.model.SecretNotFoundError as exc:
+                    logger.exception("Failed to get secret id %s: %s", secret_id, str(exc))
+                    del peer_relation.data[self.app][SECRET_ID]
+                    return None
+        else:
+            # There is no Secrets support and none relation data was created
+            # So lets create the user and store its token in the peer relation
+            secret_value = peer_relation.data[self.app].get(SECRET_KEY)
+            if secret_value:
+                return secret_value
+        return None
+
+    def _start_synapse_stats_exporter(self) -> None:
+        """Start Synapse Stats Exporter in Synapse container."""
+        container = self.unit.get_container(synapse.SYNAPSE_CONTAINER_NAME)
+        if not container.can_connect():
+            logger.error("Failed to connect to container while starting Synapse Stats Exporter")
+            return
+        self.pebble_service.replan_stats_exporter(container)
+
+    def _set_admin_access_token(self, access_token: str) -> None:
+        """Set admin access token.
+
+        Args:
+            access_token: token to save in the secret.
+        """
+        peer_relation = self._get_peer_relation()
+        if not peer_relation:
+            logger.error(
+                "Failed to get admin access token: no peer relation %s found", PEER_RELATION_NAME
+            )
+            return
+        if JUJU_HAS_SECRETS:
+            logger.debug("Adding secret")
+            secret = self.app.add_secret({SECRET_KEY: access_token})
+            peer_relation.data[self.app].update({SECRET_ID: secret.id})
+        else:
+            logger.debug("Adding peer data")
+            peer_relation.data[self.app].update({SECRET_KEY: access_token})
+
+    def _on_start(self, _: ops.HookEvent) -> None:
+        """Handle start event."""
+        if self.get_admin_access_token():
+            self._start_synapse_stats_exporter()
+            return
+        # Since Synapse is ready, the charm can create the admin access token
+        container = self.unit.get_container(synapse.SYNAPSE_CONTAINER_NAME)
+        if not container.can_connect():
+            self.unit.status = ops.MaintenanceStatus("Waiting for Synapse pebble")
```

Comment:
Not sure handling this logic only in this event will be problematic. I just tested on my machine, and randomly the event `pebble_ready` comes after the `start` event. If that happens, then the exported will not be planned.

![image](https://github.com/canonical/synapse-operator/assets/2047402/cf4a59c2-1fd4-49a9-ab9a-8d9722936580)